extensions/observability-devservices/testcontainers/src/main/java/io/quarkus/observability/testcontainers/LgtmContainer.java

```diff
@@ -140,12 +148,22 @@ public static int getPrivateOtlpPort(String otlpProtocol) {
     }
 
     private String getPrometheusConfig() {
-        Config runtimeConfig = ConfigProvider.getConfig();
-        String rootPath = runtimeConfig.getOptionalValue("quarkus.management.root-path", String.class).orElse("/q");
-        String metricsPath = runtimeConfig.getOptionalValue("quarkus.management.metrics.path", String.class).orElse("/metrics");
-        int httpPort = runtimeConfig.getOptionalValue("quarkus.http.port", Integer.class).orElse(8080); // when not set use default
-
-        return String.format(PROMETHEUS_CONFIG, config.serviceName(), rootPath, metricsPath, "host.docker.internal", httpPort);
+        String scraping = config.scrapingInterval() + "s";
+        String prometheusConfig = String.format(PROMETHEUS_CONFIG_DEFAULT, scraping);
+        if (config.forceScraping().orElse(scrapingRequired)) {
+            boolean isTest = LaunchMode.current() == LaunchMode.TEST;
+            Config runtimeConfig = ConfigProvider.getConfig();
+            String rootPath = runtimeConfig.getOptionalValue("quarkus.management.root-path", String.class).orElse("/q");
+            String metricsPath = runtimeConfig.getOptionalValue("quarkus.management.metrics.path", String.class)
+                    .orElse("/metrics");
+            String httpPortKey = isTest ? "quarkus.http.test-port" : "quarkus.http.port";
+            Optional<Integer> optionalValue = runtimeConfig.getOptionalValue(httpPortKey, Integer.class);
+            int httpPort = optionalValue.orElse(isTest ? 8081 : 8080); // when not set use default
```

Comment:
This is very confusing and brittle, but I have no better suggestions.

Reply:
Yeah, and the interesting thing is, this `orElse` actually kicks in at the end, as the prop is `Optional.empty()`.